src/libbson/src/bson/bson-utf8.c

```diff
@@ -327,17 +328,15 @@ _bson_utf8_handle_special_char (bson_unichar_t c,   /* IN */
       break;
    default: {
       // ASCII control character
-      BSON_ASSERT (c < 0x20);
+      BSON_ASSERT (c < 0x20u);
 
       const char digits[] = "0123456789abcdef";
       char codepoint[6] = "\\u0000";
 
-      codepoint[2] = digits[(c >> 12) & 0xf];
-      codepoint[3] = digits[(c >> 8) & 0xf];
```

Comment:
Were these deliberately removed?

Reply:
Sorry I forgot to add a comment about this. At this block of code `c` is between 0x01 and 0x1f (see `BSON_ASSERT (c < 0x20u);` above). So the maximum value of `c` is 0b00011111. Shifting `c` 8 or 12 bits to the right is therefore unneeded as `digits[2]` and `digits[3]` should always be 0.